File: Algorithms/Searching & Sorting/Linear Search/Linear_search.py
#Defining function for Linear Search
def Linear_Search(input_list, key):
    flag = 0
    #Iterating each item in the list and comparing it with the key searching for
    for i in range(len(input_list)):
        if(input_list[i] == key):
            #If key matches with any of the list items
            flag = 1
            print("\nKey is found in the position: {}" .format(i))
    #If key not found
    if(flag == 0):
        print("\nKey not found")

#Input List
input_list = [11,12,13,14,15,16,17,18,19]
print("List available is {}" .format(input_list))
#Getting Key from user
key = input("\nEnter key to be searched in the list")
key = int(key)
#Calling function defined to perform Linear Search
Linear_Search(input_list, key)


            
        

File: Linear Search.c
#include<stdio.h>

int LSearch(int a[],int size, int x)
{
	for(int i=0;i<size;i++)
	{
		if(a[i]==x)
		return i+1;
	}
	return -1;
}
int main()
{
	int n;
	printf("Size  : ");
	scanf("%d",&n);
	int a[n],k,i;
	for(i=0;i<n;i++)
	scanf("%d",&a[i]);
	
	printf("Value to search : ");
	scanf("%d",&k);
	printf("\n");
	
	int res=LSearch(a,n,k);
	(res==-1)?printf("Value not found"):printf("Value found at position : %d",res);
	
	return 0;
}
